pkg/sidecar/mock.go

```diff
@@ -0,0 +1,60 @@
+package sidecar
+
+import (
+	"context"
+	"errors"
+	"github.com/testground/sdk-go/sync"
+	gosync "sync"
+)
+
+// Runner
+func NewMockReactor() (Reactor, error) {
+	return &MockReactor{}, nil
+}
+
+// Reactor
+type MockReactor struct{}
+
+func (*MockReactor) Close() error { return nil }
+
+func (*MockReactor) Handle(ctx context.Context, handler InstanceHandler) error { return nil }
+
+func NewMockNetwork() *MockNetwork {
+	active := make([]string, 0)
+	configured := make([]*sync.NetworkConfig, 0)
+	mux := gosync.Mutex{}
+	return &MockNetwork{
+		Active:     active,
+		Configured: configured,
+		Closed:     false,
+		L:          &mux,
```

Comment:
1. No need for the mutex pointer here.
2. The convention is to embed the mutex by value, if you don't care about the lock/unlock methods being available publicly. Otherwise the convention is to use a variable name `lk` or `mu`, so that the lock is only available internally.